account/src/main/java/com/caldevsupplychain/account/controller/AccountController.java

```diff
@@ -0,0 +1,22 @@
+package com.caldevsupplychain.account.controller;
+
+
+import com.caldevsupplychain.common.ws.account.UserWS;
+import org.springframework.http.ResponseEntity;
+import org.springframework.validation.BindingResult;
+
+import javax.servlet.http.HttpServletRequest;
+
+
+public interface AccountController {
+
+	ResponseEntity<?> signup (String role, UserWS userWS, HttpServletRequest request, BindingResult errors);
+
+	ResponseEntity<?> activateAccount(String token);
+
+	ResponseEntity<?> login(UserWS userWS, BindingResult errors);
+
+	ResponseEntity<?> logout();
```

Comment:
We shouldn't need login/logout api. Each api is either public, or secured by permission. A good comparison is signup vs. upateUser. signup is a public API, everyone can access. updateUser is an api secured by role or permission. The only one who can updateUser is the one with correct credential(Shiro loads user context base on credential). Which means only if `contextUser.getUuid() == uuid` then updateUser is allowed.